kDriveCore/Data/UploadQueue/Servicies/FreeSpace/CacheModel.swift

```diff
@@ -0,0 +1,203 @@
+    /// The cache is a folder or a file on file system
+    case fileSystem(url: URL)
+
+    /// The cache is Kingfisher image storage on disk
+    case storageImageCache
+
+    /// Generate a collection of StorageToClean files from a source URL
+    public static func exploreFiles(for url: URL) -> [CacheItem] {
+        guard isDirectory(url: url) else {
+            #if DEBUG
+            return [CacheItem.fileSystem(url: url)]
+            #else
+            // Do not show log files in production
+            guard !url.lastPathComponent.hasSuffix(".log") else {
+                return []
+            }
+            return [CacheItem.fileSystem(url: url)]
+            #endif
+        }
+
+        // Explore children
+        let children = (try? FileManager.default.contentsOfDirectory(
+            at: url,
+            includingPropertiesForKeys: nil,
+            options: .skipsHiddenFiles
+        )) ?? [URL]()
+
+        let storages = children.reduce([CacheItem]()) { partial, child in
+            return partial + CacheItem.exploreFiles(for: child)
+        }
+        return storages
+    }
+
+    public var size: UInt64 {
+        switch self {
+        case .fileSystem(let url):
+            return getPathSize(at: url.path)
+
+        case .storageImageCache:
+            let cacheSize = try? ImageCache.default.diskStorage.totalSize()
+            return UInt64(cacheSize ?? 0)
+        }
+    }
+
+    /// Recursively explore a path and count total size
+    private func getPathSize(at path: String) -> UInt64 {
+        var fileSize = getFileSize(at: path)
+
+        // Explore children
+        if isDirectory {
+            let children = try? FileManager.default.contentsOfDirectory(atPath: path)
+            for child in children ?? [] {
+                fileSize += getPathSize(at: (path as NSString).appendingPathComponent(child))
+            }
+        }
+
+        return fileSize
+    }
+
+    /// Get the file size of a single file at path
+    private func getFileSize(at path: String) -> UInt64 {
+        var fileSize: UInt64 = 0
+        do {
+            let attributes = try FileManager.default.attributesOfItem(atPath: path)
+            if let sizeAttribute = attributes[.size] as? NSNumber {
+                fileSize = sizeAttribute.uint64Value
+            } else {
+                DDLogError("Failed to get a size attribute from path: \(path)")
+            }
+        } catch {
+            DDLogError("Failed to get file attributes for path: \(path) with error: \(error)")
+        }
+        return fileSize
+    }
+
+    var isDirectory: Bool {
+        switch self {
+        case .fileSystem(let url):
+            return Self.isDirectory(url: url)
+
+        case .storageImageCache:
+            return false
+        }
+    }
+
+    private static func isDirectory(url: URL) -> Bool {
+        var isDirectory: ObjCBool = false
+        let exists = FileManager.default.fileExists(atPath: url.path, isDirectory: &isDirectory)
```

Comment:
Didn't know this function was working like this with the `ObjCBool` !

Reply:
Maybe there is a more modern way to check this. I reused old code but wrapped it in a different way.
I'll check

Reply:
I've not found a more modern way with the canonical FileManager to check for this in a more swifty way.

Reply:
That's okay, i just didn't know about it